custom/icds/messaging/indicators.py

```diff
@@ -397,6 +383,34 @@ def get_messages(self, language_code=None):
         return messages
 
 
+def get_awws_with_old_vhnd_date(domain, user_ids):
+    return [_id for _id in user_ids if _id not in precompute_awws_in_vhnd_timeframe(domain)]
+
+
+@icds_quickcache(timeout=12 * 60 * 60, memoize_timeout=12 * 60 * 60)
+def precompute_awws_in_vhnd_timeframe(domain):
```

Comment:
I am going to add a task to trigger `precompute_awws_in_vhnd_timeframe` before the VHND SMS alert is triggered. I will first see how much time it takes to run it after [this index](https://github.com/dimagi/commcare-hq/pull/25990/files#diff-31dccf874fa0158d654334aefe7d0a45R598) is created.

Reply:
I'm not sure how to be able to tell that the task has failed, just hasn't run yet, or is still running. Would it be possible to do something like calling this directly from the indicator and using some locking to ensure only one runs at one time. Something like what you have now. 

This code: https://github.com/dimagi/commcare-hq/blob/5bc9de3f6afd009577e3226555f02e2d947da1ee/corehq/apps/app_manager/models.py#L4238-L4243 uses a `CriticalSection` to ensure that only one jad/jar file is created by first checking if there is a result when the lock is acquired. Could something similar be done here? If this value hasn't been cached by the `quickcache` decorator, then acquire a lock, when the lock is released see if another process has set the cache?

Reply:
Thanks, updated the code now.